assets/src/edit-story/components/fontPicker/fontList/index.js

```diff
@@ -0,0 +1,269 @@
+        }
+      ),
+    []
+  );
+
+  // Observe rendered font options
+  useLayoutEffect(() => {
+    const renderedOptions = optionsRef.current;
+    renderedOptions.forEach((option) => option && observer.observe(option));
+    return () => {
+      renderedOptions.forEach((option) => option && observer.unobserve(option));
+      // clear exisiting option references before next update to filteredGroup
+      optionsRef.current = [];
+    };
+  }, [observer, filteredListGroups]);
+
+  // load all seen fonts from google service
+  useEffect(() => {
+    ensureMenuFontsLoaded(
+      userSeenFonts.filter(
+        (fontName) => fontMap[fontName]?.service === 'fonts.google.com'
+      )
+    );
+  }, [ensureMenuFontsLoaded, userSeenFonts, fontMap]);
+
+  /*
+   * KEYBOARD ACCESSIBILITY
+   */
+  const filteredOptions = useMemo(() => getOptions(filteredListGroups), [
+    filteredListGroups,
+  ]);
+
+  const handleKeyPress = useCallback(
+    ({ key }) => {
+      if (key === 'Escape') {
+        onClose();
+      } else if (key === 'Enter') {
+        filteredOptions[focusIndex] &&
+          onSelect(filteredOptions[focusIndex].name);
+      } else if (key === 'ArrowUp') {
+        setFocusIndex((index) => Math.max(0, index - 1));
+      } else if (key === 'ArrowDown') {
+        setFocusIndex((index) =>
+          Math.min(filteredOptions.length - 1, index + 1)
+        );
+      }
+    },
+    [focusIndex, filteredOptions, onClose, onSelect]
+  );
+
+  useFocusOut(listRef, () => setFocusIndex(-1), []);
+
+  // scroll into view and focus focusIndex
+  useEffect(() => {
+    const listEl = listRef.current;
+    if (!listEl) {
+      return;
+    }
+    if (focusIndex === -1) {
+      listEl.scrollTo(0, 0);
+      return;
+    }
+
+    const highlighedOptionEl = optionsRef.current[focusIndex];
+    if (!highlighedOptionEl) {
+      return;
+    }
+
+    highlighedOptionEl.focus();
+    listEl.scrollTo(0, highlighedOptionEl.offsetTop - listEl.clientHeight / 2);
+  }, [focusIndex, filteredOptions, keyword]);
+
+  /*
+   * ACCESSIBILITY COMMUNICATION WITH SEARCH INPUT
+   */
+  const isExpanded = filteredOptions.length > 0;
+  useEffect(() => onExpandedChange(isExpanded), [onExpandedChange, isExpanded]);
+  useEffect(() => {
+    if (focusTrigger > 0) {
+      setFocusIndex(0);
+    }
+  }, [focusTrigger]);
+
+  return filteredOptions.length <= 0 ? (
+    <NoResult>{__('No matches found', 'web-stories')}</NoResult>
+  ) : (
+    <List
+      ref={listRef}
+      tabIndex={0}
+      id="editor-font-picker-list"
+      role="listbox"
+      onKeyDown={handleKeyPress}
+      aria-multiselectable={false}
+      aria-required={false}
+    >
+      {filteredListGroups.map(
+        (group, i) =>
+          group.options.length > 0 && (
+            <Group
+              key={group.label}
+              role="group"
+              aria-labelledby={`group-${i}`}
+            >
+              <GroupLabel id={`group-${i}`} role="presentation">
```

Comment:
I think it's better to generate this ID once above the return (in the `.map()` function) with a random unique pre-/postfix (using `uuid` as we do [here](https://github.com/google/web-stories-wp/blob/main/assets/src/edit-story/components/dropDownMenu/index.js#L56)), to make sure it's a unique ID across the entire document.